flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/plan/batch/sql/CalcTest.xml

```diff
@@ -85,7 +85,7 @@ LogicalProject(a=[$0], b=[$1], c=[$2], d=[$3])
     </Resource>
     <Resource name="planAfter">
       <![CDATA[
-Calc(select=[a, CAST(1984-07-12) AS b, CAST(14:34:24) AS c, CAST(1984-07-12 14:34:24) AS d], where=[AND(=(b, 1984-07-12), =(c, 14:34:24), =(d, 1984-07-12 14:34:24))])
+Calc(select=[a, CAST(1984-07-12) AS b, CAST(14:34:24) AS c, CAST(1984-07-12 14:34:24:TIMESTAMP(3)) AS d], where=[AND(=(b, 1984-07-12), =(c, 14:34:24), =(d, 1984-07-12 14:34:24:TIMESTAMP(3)))])
```

Comment:
I observe something weird upon comparing calcite 1.16/17/18/19 on this (not yet on 19) seems like the timestamp type representation is bouncing here and there a lot. do you think this is solely just the serialization (print string) changes or actually there's some changes to the underlying type? as this might be problematic.

Reply:
Hi @walterddr , what do you mean "timestamp type representation is bouncing here and there a lot"?

Reply:
@walterddr I think the reason is the modify of `RexLiteral.toJavaString` in https://github.com/apache/calcite/pull/1002 , The `Timestamp` default precision is 3, I think it hasn't changed.

Reply:
looks like it. what I meant is that I've been changing the test cases string literals back and forth for some of the pass calcite upgrades. Seems like not an issue then.